utils/navigate.py

```diff
@@ -0,0 +1,54 @@
+from utils.log import logger
+
+
+class Navigate(object):
+    def __init__(self):
+        self.cache = {}
+
+    def register(self, cls):
```

Comment:
using `[cls, obj.__name__]' should be able to safe some spaces

for extra measure the decorator perhaps should support giving a different name as a option